dashboard/src/utils/records.ts

```diff
@@ -0,0 +1,24 @@
+const isRecord = (obj: unknown): obj is Record<string, unknown> => {
+  return typeof obj === 'object' && !Array.isArray(obj);
+};
```

Comment:
You might want to add
```suggestion
const isRecord = (obj: unknown): obj is Record<string, unknown> => {
  return typeof obj === 'object' && obj !== null && !Array.isArray(obj);
};
```
Unless you want null to be considered a record. Also, we already have the function `isStringRecord` in utils.ts. Please remove it from there and substitute the occurrences of `isStringRecord` to `isRecord`